security/rbac/endpoint_enforcer.go

```diff
@@ -0,0 +1,50 @@
+package rbac
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gaia-pipeline/gaia"
+)
+
+// EndpointEnforcer represents the interface for enforcing RBAC using the echo.Context.
+type EndpointEnforcer interface {
+	Enforce(username, method, path string, params map[string]string) (bool, error)
+}
+
+// Enforce uses the echo.Context to enforce RBAC. Uses the rbacapiMappings to apply policies to specific endpoints.
+func (e *enforcerService) Enforce(username, method, path string, params map[string]string) (bool, error) {
+	group := e.rbacapiMappings
+
+	endpoint, ok := group.Endpoints[path]
+	if !ok {
+		gaia.Cfg.Logger.Warn("path not mapped to api group", "path", path)
+		return true, nil
+	}
+
+	perm, ok := endpoint.Methods[method]
+	if !ok {
+		gaia.Cfg.Logger.Warn("method not mapped to api group path", "path", path, "method", method)
+		return true, nil
+	}
+
+	splitAction := strings.Split(perm, "/")
+	namespace := splitAction[0]
+	action := splitAction[1]
+
+	fullResource := "*"
```

Comment:
So we need to use `*` for endpoints where there isn't a resource id specified in the api-mappings yaml. Right now that is something like `secrets/create` that doesn't need the enforcement of a resource id on it.

Reply:
Wait, why isn't there a secrets and why doesn't that need enforcement? :D

Reply:
I think you have misunderstood mate! We have permissions on secrets. I just mean that `secrets/create` doesn't have resource (a key ID) enforced because you haven't created something yet!

So:
`not specified` < cannot create secrets
`p, secrets, create, *, deny` < cannot create secrets
`p, secrets, create, *, allow` < can create secrets